stag-library-compiler/src/main/java/com/vimeo/stag/processor/generators/model/accessor/MethodFieldAccessor.java

```diff
@@ -92,16 +106,53 @@ private static String findSetterMethodName(@NotNull VariableElement variableElem
         throw new UnsupportedOperationException("Unable to find setter for variable");
     }
 
+    private static boolean isBoolean(@NotNull TypeMirror typeMirror) {
+        return TypeUtils.areEqual(TypeUtils.getPrimitive(typeMirror.getKind()), TypeUtils.getPrimitive(TypeKind.BOOLEAN))
+            || TypeUtils.areEqual(typeMirror, ElementUtils.getTypeFromClass(Boolean.class));
+    }
+
+    private static boolean isSupportedGetter(@NotNull ExecutableElement method,
+                                             @NotNull VariableElement variableElement,
+                                             @NotNull Notation namingNotation) {
+        final String methodName = method.getSimpleName().toString();
+        final TypeMirror returnType = method.getReturnType();
+        final String variableNameMethodComponent = getVariableNameAsMethodName(variableElement, namingNotation);
+
+        final boolean variableNameBeginsWithIs = variableNameBeginsWithIs(variableNameMethodComponent);
+
+        return (isBoolean(returnType)
+            && ((!variableNameBeginsWithIs && methodName.equals("is" + variableNameMethodComponent))
+            || (variableNameBeginsWithIs && methodName.equals(convertCharAtToLowerCase(variableNameMethodComponent, 0)))))
+            || methodName.equals("get" + variableNameMethodComponent);
+    }
+
+    private static boolean variableNameBeginsWithIs(@NotNull String variableName) {
+        return variableName.length() > 2
+            && (variableName.startsWith("is") || variableName.startsWith("Is"))
+            && Character.isUpperCase(variableName.charAt(2));
+    }
+
+    @NotNull
+    private static String convertCharAtToLowerCase(@NotNull String string, int index) {
```

Comment:
Minor: This could probably be moved into a `StringUtils` class.  Maybe the above method too if you take `is` as a param?